br/pkg/lightning/config/global.go

```diff
@@ -273,6 +291,16 @@ func LoadGlobalConfig(args []string, extraFlags func(*flag.FlagSet)) (*GlobalCon
 	if len(filter) > 0 {
 		cfg.Mydumper.Filter = filter
 	}
+	if *checkOnlyCfg != "" {
+		if cfg.App.ServerMode {
+			return nil, errors.New(`cannot set "check-only" if server-mode is enabled`)
```

Comment:
why is the check-only config interfering with server-mode?

Reply:
check-only is used on command line only, check, print result and exit

Reply:
this does not sound like a blocker requiring check-only and server-mode to be mutually exclusive. in server mode we could "submit a task → check → print result → back to idle" too

Reply:
check-only is used on command line only right now, maybe we can do it later